File: ainpc-core-plugin/src/main/java/ro/ainpc/gui/GuiSession.java
package ro.ainpc.gui;

import java.util.Map;
import java.util.UUID;

public class GuiSession {

    private final UUID sessionId;
    private final UUID playerId;
    private final GuiKey key;
    private final long openedAt;
    private Map<Integer, GuiButton> buttons = Map.of();

    public GuiSession(UUID sessionId, UUID playerId, GuiKey key, long openedAt) {
        this.sessionId = sessionId;
        this.playerId = playerId;
        this.key = key;
        this.openedAt = openedAt;
    }

    public UUID getSessionId() {
        return sessionId;
    }

    public UUID getPlayerId() {
        return playerId;
    }

    public GuiKey getKey() {
        return key;
    }

    public long getOpenedAt() {
        return openedAt;
    }

    public Map<Integer, GuiButton> getButtons() {
        return buttons;
    }

    public void setButtons(Map<Integer, GuiButton> buttons) {
        this.buttons = Map.copyOf(buttons != null ? buttons : Map.of());
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/gui/GuiSession.kt
package ro.ainpc.gui

import java.util.UUID

class GuiSession(
    sessionId: UUID,
    playerId: UUID,
    key: GuiKey,
    openedAt: Long
) {
    private val sessionIdValue = sessionId
    private val playerIdValue = playerId
    private val keyValue = key
    private val openedAtValue = openedAt
    private var buttonsValue: Map<Int, GuiButton> = emptyMap()

    fun getSessionId(): UUID = sessionIdValue

    fun getPlayerId(): UUID = playerIdValue

    fun getKey(): GuiKey = keyValue

    fun getOpenedAt(): Long = openedAtValue

    fun getButtons(): Map<Int, GuiButton> = buttonsValue

    fun setButtons(buttons: Map<Int, GuiButton>?) {
        buttonsValue = java.util.Map.copyOf(buttons ?: emptyMap())
    }
}
